src/bios.rs

```diff
@@ -254,31 +254,30 @@ unsafe fn advertise_svsm_presence(bios_info: &mut BiosInfo, caa: PhysAddr) -> bo
         return false;
     }
 
-    let bios_secrets_pa: PhysAddr = PhysAddr::new(section.address_u64());
-
-    let bios_secrets_va: VirtAddr =
-        match pgtable_map_pages_private(bios_secrets_pa, section.size_u64()) {
-            Ok(v) => v,
-            Err(_e) => return false,
-        };
-    let svsm_secrets_va: VirtAddr = get_svsm_secrets_page();
-
-    // Copy the Secrets page to the BIOS Secrets page location
-    let bios_secrets: *mut SnpSecrets = bios_secrets_va.as_mut_ptr();
-    let svsm_secrets: *const SnpSecrets = svsm_secrets_va.as_ptr();
-    *bios_secrets = *svsm_secrets;
-
-    // Clear the VMPCK0 key
-    (*bios_secrets).clear_vmpck0();
-
-    // Advertise the SVSM
-    (*bios_secrets).set_svsm_base(pgtable_va_to_pa(get_svsm_begin()).as_u64());
-    (*bios_secrets).set_svsm_size(get_svsm_end().as_u64() - get_svsm_begin().as_u64());
-    (*bios_secrets).set_svsm_caa(caa.as_u64());
-    (*bios_secrets).set_svsm_max_version(1);
-    (*bios_secrets).set_svsm_guest_vmpl(1);
-
-    pgtable_unmap_pages(bios_secrets_va, section.size_u64());
+    {
```

Comment:
Here it's probably OK to keep both pages mapped until the end of the function (or any early return). I'll change it.

In other places, this might not be the case. For example in `handle_create_vcpu_request()` we need to unmap the vmsa page and then call `invlpgb_all()` and `tlbsync()`. If we don't put the explicit brackets, the drop of the MapGuard will occur at the end of the function *after* `invlpgb_all()` and `tlbsync()`, modifying the behaviour. If we want to avoid the extra indent we can extract the code that deals with the mapped page to a helper function, but that might be cumbersome if there are a lot of stack argument we need to pass to that function.